quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/update/Quarkus32to38CliUpdateIT.java

```diff
@@ -0,0 +1,263 @@
+        quarkusCLIAppManager.updateApp(app);
+
+        Map<String, String> renames = new HashMap<>();
+        // Quarkus 3.5
+        renames.put("uniMemoize.atLeast", "uniMemoize.forFixedDuration");
+        renames.put("uniAndGroup.combinedWith", "uniAndGroup.with");
+
+        // Quarkus 3.7
+        renames.put("logRecord.getThreadID", "logRecord.getLongThreadID");
+        renames.put("logRecord.setThreadID", "logRecord.setLongThreadID");
+
+        checkRenamesInFile(app.getFileFromApplication("src/main/java/org/acme", "MethodRenameResource.java"), renames);
+
+        app.start();
+
+        assertEquals("Hello Uni", app.given().get("/rename/uni/memoize").body().asString());
+        assertEquals("Hello group", app.given().get("/rename/uni/group").body().asString());
+        assertEquals("42", app.given().get("/rename/thread").body().asString());
+    }
+
+    @Test
+    public void packageNameChangeTest() throws IOException {
+        QuarkusCliRestService app = cliClient.createApplicationFromExistingSources("app", null, RENAME_PACKAGE_APP);
+        quarkusCLIAppManager.updateApp(app);
+
+        Map<String, String> renames = new HashMap<>();
+        // Quarkus 3.5
+        renames.put("org.hibernate.search.mapper.orm.coordination.outboxpolling.OutboxPollingExtension",
+                "org.hibernate.search.mapper.orm.outboxpolling.OutboxPollingExtension");
+        renames.put("javax.security.cert.Certificate", "java.security.cert.Certificate");
+
+        checkRenamesInFile(app.getFileFromApplication("src/main/java/org/acme", "PackageChangeResource.java"), renames);
+
+        app.start();
+
+        assertEquals("Hello cert", app.given().get("/rename/cert").body().asString());
+        assertEquals("OutboxPollingExtension", app.given().get("/rename/extension").body().asString());
+    }
+
+    @Test
+    public void updateMultiModuleAppTest() throws XmlPullParserException, IOException {
+        QuarkusCliRestService app = cliClient.createApplicationFromExistingSources("app", null, MULTI_MODULE_APP);
+        quarkusCLIAppManager.updateApp(app);
+
+        // parent
+        assertTrue(getPom(app).getProperties().getProperty("quarkus.platform.version").startsWith("3.8"),
+                "Parent project should be updated to quarkus 3.8.x");
+
+        // rest-client (module)
+        Dependency restClientDependency = new QuarkusDependency("io.quarkus:quarkus-resteasy-client-jackson");
+        assertTrue(getPom(app, "rest-client").getDependencies().contains(restClientDependency),
+                "quarkus-rest-client-jackson should be updated to quarkus-resteasy-client-jackson in child project");
+
+        // TODO: it would be good to actually run the updated app, but FW currently cannot run multi module app
+    }
+
+    /**
+     * Update resteasy app, while also upgrading it from java 11 to 17
+     */
+    @Test
+    public void updateAndRunApp() throws XmlPullParserException, IOException {
```

Comment:
`old` app (the one you are updating) uses RHBQ, it would be nice to check that new one is still RHBQ.

Reply:
Thanks for long answer, maybe I am slow but I need little more context:

> This is kind of difficult to do, since it presumes that quarkus CLI, which will do the operations, is configured to run with RHBQ. Meaning even if it receives just 3.8.5 as a parameter it will work with RHBQ 3.8 stream, not with upstream quarkus. And if it receives anything else than specified version (in this case 3.8.*) it will not run at all.

I thought based on this comment https://github.com/quarkusio/quarkus/issues/36777#issuecomment-1831709951 that unless you specify `--platform` or `--stream` then neither group id or stream is going to change. What I'd like to do is to check that group id didn't change, hence RHBQ is still used so that we know that we accidentally didn't update customer to community. Which was my motivation for this comment. What is stopping you?

> Test work a bit differently if they receive just 3.8.5 as a parameter or 3.8.5-redhat-. If they receive specific RH build, it will go for that exact BOM, if they receive just version, they will go for a stream. Either way, they should work with RHBQ.

Either way, I'd expect that update app is still Red Hat build of Quarkus regardless of the registry. Do you mean that if config is like

```
---
registries:
- registry.quarkus.io
- registry.quarkus.redhat.com
```

then I'll get community app instead? If so, sure, I'd test that depending on https://github.com/quarkus-qe/quarkus-test-framework/blob/main/quarkus-test-core/src/main/java/io/quarkus/test/services/quarkus/model/QuarkusProperties.java#L35 you are able to tell what is expected result. Because if Quarkus version is RHBQ, you are likely to test it with RH registry. Nevertheless maybe locally not...

Just to be clear, this is no biggie, if there is an issue with checking this, I don't require it.

Reply:
Great, thanks.

Reply:
added